### PR TITLE
feat: Unify attribute lookups across span and scope traits

### LINKED ISSUE #160: Split OTLP log attributes
[discussion](https://github.com/triplecloudtech/trust/pull/14#discussion_r3573388764)

Icegate currently stores resource, scope, and log attributes in a single merged `attributes` map and normalizes key names. This loses attribute ownership, can cause key collisions, and forces Trust to reconstruct the OTLP structure.

Update log ingestion and storage:

- store `resource_attributes`, `scope_attributes`, and `log_attributes` separately;
- preserve canonical OTEL key names, including dots (`service.name`, `http.method`, etc.);
- retain `service_name`, `scope_name`, and `scope_version` where needed for query optimization;
- define schema evolution and backward compatibility for existing records;
- update Flight SQL and Loki/LogQL read paths;
- prevent attributes from different scopes being mixed during filtering, label extraction, or result construction;
- add tests for duplicate keys across scopes, dotted OTEL names, and legacy data.

Acceptance criteria:

1. The three attribute groups are stored independently.
2. Identical keys in different groups do not overwrite one another.
3. Flight SQL exposes separate attribute groups.
4. Existing Loki/LogQL behavior remains compatible.
5. Legacy records remain readable through migration or fallback.

### PR DESCRIPTION
**Pull request by frisbeeman**
## Summary by CodeRabbit

- **New Features**
  - OTLP resource, scope, span, log, and data-point attributes are now handled separately for more accurate querying and display.
  - Loki label discovery and values support dotted attribute names with normalized wire-format output.
  - Tempo and TraceQL now support instrumentation-scope attributes through span queries, including precedence when names overlap.
  - LogQL supports `ip` as a label name and no longer accepts legacy colon-prefixed labels.

- **Bug Fixes**
  - Improved attribute filtering, grouping, aggregation, and null handling across queries.
  - Updated schema documentation and migration guidance for the new attribute columns.

### CODE REVIEW

File: crates/icegate-query/src/loki/executor.rs [hunk 1]
```diff
@@ -336,19 +355,32 @@ impl QueryExecutor {
...
-            query_ctx.start,
-            query_ctx.end,
-            &super::LOGS_METADATA_CONFIG,
-            extra,
-        )
-        .await
-        .map_err(|e| LokiError(QueryError::from(e)))?;
+        let mut labels: BTreeSet<String> = BTreeSet::new();
+        for config in &super::LOGS_METADATA_CONFIGS {
+            let scanned = crate::engine::metadata_scan::scan_labels(
```
Comment (s-prosvirnin) [inline crates/icegate-query/src/loki/executor.rs]: we have three similar read of metadata and parquet here. Can we make only one read? Same in `execute_label_values`

File: crates/icegate-common/src/schema.rs [hunk 2]
```diff
@@ -1228,8 +1329,8 @@ pub fn events_sort_order(schema: &Schema) -> Result<SortOrder> {
 /// - `timestamp` (descending) - recent-first ordering
 #[allow(clippy::too_many_lines)]
 pub fn metrics_schema() -> Result<Schema> {
-    // Create Map<String, String> for main attributes (field IDs: 32, 33)
-    let attributes_map = Type::Map(MapType::new(
+    // Create Map<String, String> for data-point attributes (field IDs: 32, 33)
+    let data_point_attributes_map = Type::Map(MapType::new(
```
Comment (s-prosvirnin) [inline crates/icegate-common/src/schema.rs:1333]: Do we need new IDs here? Will there be a conflict?

File: crates/icegate-ingest/src/transform/logs.rs [hunk 3]
```diff
@@ -214,12 +248,12 @@ pub fn logs_to_record_batch(
...
-/// resource flattening to keep the attributes MAP free of redundant copies
-/// — both to shrink Parquet pages and to keep the row-group key dictionary
-/// uncluttered.
-const LOG_PROMOTED_RESOURCE_KEYS: &[&str] = &[COL_SERVICE_NAME];
+/// Resource-attribute keys, in their OTLP dotted form, that already have a
+/// dedicated top-level column on the logs table. Dropped from the resource
+/// level so the MAP holds no redundant copy — smaller Parquet pages and a
+/// tidier row-group key dictionary. A more specific level may still supply
+/// the same key as a deliberate override.
+const LOG_PROMOTED_RESOURCE_KEYS: &[&str] = &["service.name"];
```
Comment (s-prosvirnin) [inline crates/icegate-ingest/src/transform/logs.rs]: `COL_SERVICE_NAME`?

File: crates/icegate-query/src/logql/datafusion/udf/map_merge_normalized.rs [hunk 4]
```diff
@@ -0,0 +1,385 @@
...
+/// apply, and it has to stay that way for the reason given above.
+///
+/// Series identity, `by`/`without` grouping, and key serialization all operate
+/// on one map, so merging here keeps that logic identical to the pre-split
+/// version instead of triplicating it.
+///
+/// Output keys are sorted: grouping serializes them into a string, so a stable
+/// order is required for two identical label sets to group together.
+#[derive(Debug, PartialEq, Eq, Hash)]
+pub struct MapMergeNormalized {
```
Comment (s-prosvirnin) [inline crates/icegate-query/src/logql/datafusion/udf/map_merge_normalized.rs]: Maybe rename? The function is specific to attributes

File: crates/icegate-common/src/schema.rs (hunk 2 above)
Comment (frisbeeman) [reply crates/icegate-common/src/schema.rs:1333]: yes, just recreate db

**Review by s-prosvirnin — APPROVED** [review]
